jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -313,13 +315,26 @@ def _update_mark_scale(self, *args):
         else:
             self.background.scale_factor = self.slice_spectral_value/self.reference_spectral_value
 
+    def _on_gloabl_display_unit_changed(self, msg={}):
+
+        if msg.axis == 'spectral_y':
+            self.spectrum_y_units = str(msg.unit)
+
+        # a 'flux' and 'sb' message should be recieved back to back from
+        # the unit conversion plugin, so don't need to sync them immediatley
+        # within each message recieved
+        if msg.axis == 'flux':
+            self.flux_unit = str(msg.unit)
+        if msg.axis == 'sb':
+            self.sb_unit = str(msg.unit)
+
```

Comment:
this then needs to also make a call to `_update_results_units`/`_update_units_on_function_selection` (in which case maybe it shouldn't be renamed), right?
```suggestion
else:
    return
self._update_units_on_function_selection(msg)
```

Reply:
I don't think  _update_units_on_function_selection needs to be called here. if the flux display unit is changed, the 'flux_unit' and 'sb_unit' traitlet will be updated. the function selection determines which one of these becomes the display unit, but both are available and updated to the new unit

Reply:
i don't think this _update_units_on_function_selection needs to be called again, but correct me if i'm wrong.

upon selection of a new 'flux' unit in UC, both the flux_unit and sb_unit traitlets will be updated in the spectral extraction plugin. which of these is used for the display unit is determined by choice of 'function', but it has access to both which are synced with the current selection in unit conversion now.

Reply:
But those traitlets aren't observed in that method and the string itself is set, not a reference. I can test in the morning to see what actually happens though, I only had time to skim the code.

Reply:
Tested and this does not currently work, but didn't work in main either and is technically covered by another ticket, so probably can be deferred until then so we can get the rest of this in asap.